python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1514,15 +1521,18 @@ def do_evaluate(cls, name: str, options: Any, df: DataFrame) -> DataFrame:
                         plc.interop.from_arrow(
                             pa.array(
                                 pivotees,
-                                type=plc.interop.to_arrow(variable_dtype),
+                                type=plc.interop.to_arrow(schema[variable_name]),
```

Comment:
Why, is it you need the schema elsewhere?

Reply:
We discussed offline, but for ref: We cannot properly reconstruct the object if the structure of its constructor arguments change.